relay/blockchain/currency_network_proxy.py

```diff
@@ -198,3 +198,16 @@ def get_all_network_events(self,
 
     def estimate_gas_for_transfer(self, sender, receiver, value, max_fee, path):
         return self._proxy.estimateGas({'from': sender}).transfer(receiver, value, max_fee, path)
+
+    def estimate_gas_for_payment_path(self, payment_path):
+        if not payment_path.path:
+            return 0
+        source = payment_path.path[0]
+        target = payment_path.path[-1]
+        try:
+            return self._proxy.estimateGas({'from': source}).transfer(target,
+                                                                      payment_path.value,
+                                                                      payment_path.fee,
+                                                                      payment_path.path[1:])
+        except ValueError as e:  # should mean out of gas, so path was not right.
+            return 0
```

Comment:
I dont think is is a good idea to handle that error with returning 0. This is hiding the error and the value is confusing. It is out of gas so if we would return a value the suitable would be infinity. I think it is better to throw an exception like InvalidPathException or OutOfGasException

Reply:
Well, catching the error is exactly why I've introduced this function. Returning 0 is also consistent with what we do in our REST API. I can understand your argument, especially given the fact that this has been moved to another file. I can move it back to resources.py as a helper function or add a comment explaining it.
Raising another error doesn't help the code in resources calling this method.

Reply:
No it is not the same. What we do in the other function only makes sense because we return an empty path which means that no path was found, so no payment can be made. Here we return the value for a certain path, which is totally different. 0 with empty path is consistent, 0 with a path that throws is not.

Reply:
I think raising an error would make it clear and in the calling code you could just catch it and do the same logic where there is now the == 0 check.

Reply:
> What we do in the other function only makes sense because we return an empty path which means that no path was found, so no payment can be made

This is just how you interpret that empty path, i.e.you give a special meaning to that return value.

We could as well raise an error instead of returning a special value in the REST API.

Anyway,I've moved the exception catching to the calling function in 50dfea897dd67d310c0d74fb55c42.